frontend/src/metabase/admin/permissions/selectors/confirmations.ts

```diff
@@ -123,20 +131,45 @@ export function getControlledDatabaseWarningModal(
   entityId: EntityId,
 ) {
   if (currDbPermissionValue !== DataPermissionValue.CONTROLLED) {
-    const [entityType, entityTypePlural] = isTableEntityId(entityId)
-      ? [t`table`, t`tables`]
-      : isSchemaEntityId(entityId)
-      ? [t`schema`, t`schemas`]
-      : [t`entity`, t`entities`];
+    const [entityType, entityTypePlural] = getEntityTypeFromId(entityId);
     return {
-      title: t`Change access to this database to granular?`,
+      title: t`Change access to this database to “Granular”?`,
       message: t`Just letting you know that changing the permission setting on this ${entityType} will also update the database permission setting to “Granular” to reflect that some of the database’s ${entityTypePlural} have different permission settings.`,
```

Comment:
Will this localize to "this tables" in certain cases"?

Perhaps this calls for ttag pluralization [like this](https://github.com/metabase/metabase/blob/7d19cb3/frontend/src/metabase/query_builder/components/view/QuestionRowCount/QuestionRowCount.tsx#L148)?

Reply:
I don't think so because there's the separate `entityTypePlural` variable.